src/app/features/container/container.tsx

```diff
@@ -1,25 +1,63 @@
-import { useEffect } from 'react';
-import { Outlet, useLocation } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import { Outlet, useLocation, useNavigate } from 'react-router-dom';
 
+import { ChainID } from '@stacks/transactions';
+import { OnboardingSelectors } from '@tests/selectors/onboarding.selectors';
+import { SettingsSelectors } from '@tests/selectors/settings.selectors';
+import { Box } from 'leather-styles/jsx';
+import { token } from 'leather-styles/tokens';
+
+import { RouteUrls } from '@shared/route-urls';
 import { closeWindow } from '@shared/utils';
 
 import { useAnalytics, useInitalizeAnalytics } from '@app/common/hooks/analytics/use-analytics';
 import { LoadingSpinner } from '@app/components/loading-spinner';
+import { CurrentAccountAvatar } from '@app/features/current-account/current-account-avatar';
+import { CurrentAccountName } from '@app/features/current-account/current-account-name';
+import { SwitchAccountDialog } from '@app/features/dialogs/switch-account-dialog/switch-account-dialog';
+import { InAppMessages } from '@app/features/hiro-messages/in-app-messages';
 import { useOnSignOut } from '@app/routes/hooks/use-on-sign-out';
 import { useOnWalletLock } from '@app/routes/hooks/use-on-wallet-lock';
 import { useHasStateRehydrated } from '@app/store';
-import { useRouteHeaderState } from '@app/store/ui/ui.hooks';
+import { useCurrentNetworkState } from '@app/store/networks/networks.hooks';
+import { ContainerLayout } from '@app/ui/components/containers/container.layout';
+import { NetworkModeBadge } from '@app/ui/components/containers/headers/components/network-mode-badge';
+import { Header } from '@app/ui/components/containers/headers/header';
+import { Flag } from '@app/ui/components/flag/flag';
+import { Logo } from '@app/ui/components/logo';
+import { HamburgerIcon } from '@app/ui/icons/';
 
 import { useRestoreFormState } from '../popup-send-form-restoration/use-restore-form-state';
-import { SettingsDropdown } from '../settings-dropdown/settings-dropdown';
-import { SwitchAccountDrawer } from '../switch-account-drawer/switch-account-drawer';
-import { ContainerLayout } from './container.layout';
+import { Settings } from '../settings/settings';
+import { TotalBalance } from './total-balance';
+import {
+  getDisplayAddresssBalanceOf,
+  isKnownPopupRoute,
+  isRpcRoute,
+  showAccountInfo,
+  showBalanceInfo,
+} from './utils/get-popup-header';
+import { getTitleFromUrl } from './utils/get-title-from-url';
+import {
+  canGoBack,
+  getIsSessionLocked,
+  getPageVariant,
+  hideLogo,
+  hideSettingsOnSm,
+  isLandingPage,
+  isNoHeaderPopup,
+  isSummaryPage,
+} from './utils/route-helpers';
 
 export function Container() {
```

Comment:
I know this was here before, but wondering why `Container` is a feature looking at where you've moved `container.layout`?

Reply:
It shouldn't be a feature, you're right. I should move it somewhere else, maybe alongside `App.tsx` at the top level? It's really not a specific feature at all, it's everything.

This is more an smart container that ties together the logic needed to build all of our pages. 

I moved `container.layout` into UI as its dumb - presentation only, and I was hoping to use it to help build composed storybook stories so I can have visual regression tests on full pages.

Reply:
@kyranjamie do you know why `Container` is located in our features folder?

Reply:
The container component handles business logic, such as the onLock/onSignout hooks, and reads global app/route state. IMO this makes it a smart component, which in our dir structure fits into the feature folder. Maybe this naming is off, but I see it as a kind of "root feature".

Reply:
Agree, lets leave it here. It seems unusual as a `feature` but it's better there than floating on its own at the root level beside `app.tsx`